api/libretimeapi/models/schedule.py

```diff
@@ -24,26 +24,58 @@ def get_owner(self):
 
     def get_cueout(self):
         """
-        Returns a cueout that is based on the current show. Cueout of a specific
-        item can potentially overrun the show that it is scheduled in. In that
-        case, the cueout should be the end of the show. This prevents the next
-        show having overlapping items playing.
+        Returns a scheduled item cueout that is based on the current show instance.
+
+        Cueout of a specific item can potentially overrun the show that it is
+        scheduled in. In that case, the cueout should be the end of the show.
+        This prevents the next show having overlapping items playing.
+
+        Cases:
+        - When the schedule ends before the end of the show instance,
+        return the stored cueout.
+
+        - When the schedule starts before the end of the show instance
+        and ends after the show instance ends,
+        return timedelta between schedule starts and show instance ends.
+
+        - When the schedule starts after the end of the show instance,
+        return the stored cue_out even if the schedule WILL NOT BE PLAYED.
         """
-        if self.instance.ends < self.ends:
+        if self.starts < self.instance.ends and self.instance.ends < self.ends:
```

Comment:
Now that we filter invalid items out, should we add a note here ? And is that check really useful ?